OpenRA.Mods.Common/Lint/CheckActors.cs

```diff
@@ -18,10 +19,16 @@ public class CheckActors : ILintMapPass
 	{
 		public void Run(Action<string> emitError, Action<string> emitWarning, ModData modData, Map map)
 		{
-			var actorTypes = map.ActorDefinitions.Select(a => a.Value.Value);
-			foreach (var actor in actorTypes)
-				if (!map.Rules.Actors.Keys.Contains(actor.ToLowerInvariant()))
-					emitError($"Actor `{actor}` is not defined by any rule.");
+			var scriptBindings = Game.ModData.ObjectCreator.GetTypesImplementing<ScriptGlobal>().SelectMany(t => Utility.GetCustomAttributes<ScriptGlobalAttribute>(t, true)).ToArray();
+			foreach (var actor in map.ActorDefinitions)
+			{
+				var name = actor.Value.Value;
+				if (!map.Rules.Actors.Keys.Contains(name.ToLowerInvariant()))
```

Comment:
Should this be `ContainsKey`? (previous code but since we're here)

Reply:
Is there a difference? Will change to `ContainsKey` regardless since it feels more natural.

Reply:
`ContainsKey` does a fast lookup in the dictionary. `Keys.Contains` does a slow linear search over the keys collection.